sospice/catalog/catalog.py

```diff
@@ -451,3 +453,53 @@ def plot_fov(self, ax, **kwargs):
                 fontsize="xx-large",
                 borderaxespad=2,
             )
+
+    def download_files(
+        self,
+        base_dir,
+        base_url=None,
+        keep_tree=True,
+        downloader=None,
+        max_download=1000,
+    ):
+        """
+        Download all files from Catalog.,
+
+        Parameters
+        ----------
+        base_dir: Path or str
+            Base directory to download file to
+        base_url: str
+            Base URL for file
+        keep_tree: bool
+            Keep tree directory structure (by level and date)
+        downloader: parfive.Downloader
+            If provided, enqueue file for download instead of downloading it.
+            To download enqueued files, run `downloader.download()`
+        max_download: int
+            default maximum of 1000 files can be downloaded.
+            User can override it by changing the value.
+
+        Return
+        ------
+        parfive.Result
+            Download result (or None if file has only been enqueued)
+        """
+        get_catalog = self.read_catalog()
+        if max_download > 1000:
+            warnings.warn(
+                "You are overriding the default max_download: This might cause performance issues.",
+                UserWarning,
+            )
+
+        processed_downloads = get_catalog.iloc[:max_download].apply(
+            lambda row: FileMetadata(row).download_file(
+                base_dir=base_dir,
+                base_url=base_url,
+                keep_tree=keep_tree,
+                downloader=downloader,
+            ),
+            axis=1,
+        )
+
```

Comment:
...and here, do the actual download, if no `Download` object has been provided when calling this function (condition on `do_download`, to be defined above)